### ISSUE TITLE
Speed up min builds

### ISSUE THREAD

**Issue by zpao** [facebook/react#1933]
We're minifiying twice thanks to `uglifyify`. We need this to remove dead requires. But it also doesn't minify all the way, so we still need `uglify` as well. I haven't actually looked to see if there's a better way.

**Comment by tgriesser** [comment]
@spicyj is right that webpack is not only smart enough to strip out dead requires, but it strips out dead code paths entirely, so in your build (pre-minified code) instead of:

```js
("production" !== "development" ? invariant(
    children.nodeType !== 1,
    'traverseAllChildren(...): Encountered an invalid child; DOM ' +
    'elements are not valid children of React components.'
) : invariant(children.nodeType !== 1));
```
you just get the much nicer:
```js
invariant(children.nodeType !== 1,
  'traverseAllChildren(...): Encountered an invalid child; DOM ' +
  'elements are not valid children of React components.'
);
```

**Comment by zertosh** [comment]
wow I totally missed your response @zpao. I just pushed an updated version of `unreachable-branch-transform` that uses `recast` instead of `jstransform`, that doesn't have that problem. I did a quick time benchmark between this vs an extra uglify pass, that the time difference was negligible.

Though, I was able to shave off 2 sec by turning off `mangle` and a few compress options on the first pass of `uglifyify` - since it's only the dead code removal that matters. For the 2 sec improvement (prob less on faster machines), it wasn't worth PRing the changes to the grunt task that allow passing options to transforms :disappointed:

The real killer is `derequire`.
